### DISCUSSION TITLE
FPS problem

### DISCUSSION THREAD

**Discussion by pangr** [STVIR/pysot#181] [General]
when I test the code in P4 GPU， I found the FPS is lower than  PySOT Model Zoo （r50 only have 20 fps）.Is anyone have same problem?

**Comment by ZhiyuanChen** [comment]
I couldn't understand your problem
What do you mean by PySot model zoo?

**Comment by vhnamgau** [comment]
In your document: FPS ~ 80 FPS
I have the same problem with GTX 2080Ti  -> 15.37 frame per second. 

In tools/demo.py, we add count and start_time variable to calculate FPS
    # build tracker
    tracker = build_tracker(model)

    first_frame = True
    if args.video_name:
        video_name = args.video_name.split('/')[-1].split('.')[0]
    else:
        video_name = 'webcam'
    cv2.namedWindow(video_name, cv2.WND_PROP_FULLSCREEN)
    start_time = time.time()
    count = 0
    for frame in get_frames(args.video_name):
        count += 1
        if first_frame:
            try:
                init_rect = cv2.selectROI(video_name, frame, False, False)
            except:
                exit()
            tracker.init(frame, init_rect)
            first_frame = False
        else:
            outputs = tracker.track(frame)
            if 'polygon' in outputs:
                polygon = np.array(outputs['polygon']).astype(np.int32)
                cv2.polylines(frame, [polygon.reshape((-1, 1, 2))],
                              True, (0, 255, 0), 3)
                mask = ((outputs['mask'] > cfg.TRACK.MASK_THERSHOLD) * 255)
                mask = mask.astype(np.uint8)
                mask = np.stack([mask, mask*255, mask]).transpose(1, 2, 0)
                frame = cv2.addWeighted(frame, 0.77, mask, 0.23, -1)
            else:
                bbox = list(map(int, outputs['bbox']))
                cv2.rectangle(frame, (bbox[0], bbox[1]),
                              (bbox[0]+bbox[2], bbox[1]+bbox[3]),
                              (0, 255, 0), 3)
            #cv2.imshow(video_name, frame)
            #cv2.waitKey(40)

    print(count / (time.time() - start_time))

**Comment by buaacarzp** [comment]
您好，这个FPS是不是跟图像的大小有关系？ 我看官方好像没有如何去测试FPS的代码，vot中的图像的尺寸是一样大的吗？

**Comment by ZhiyuanChen** [comment]
> In your document: FPS ~ 80 FPS
> I have the same problem with GTX 2080Ti -> 15.37 frame per second.
> 
> In tools/demo.py, we add count and start_time variable to calculate FPS
> # build tracker
> tracker = build_tracker(model)
> 
> ```
> first_frame = True
> if args.video_name:
>     video_name = args.video_name.split('/')[-1].split('.')[0]
> else:
>     video_name = 'webcam'
> cv2.namedWindow(video_name, cv2.WND_PROP_FULLSCREEN)
> start_time = time.time()
> count = 0
> for frame in get_frames(args.video_name):
>     count += 1
>     if first_frame:
>         try:
>             init_rect = cv2.selectROI(video_name, frame, False, False)
>         except:
>             exit()
>         tracker.init(frame, init_rect)
>         first_frame = False
>     else:
>         outputs = tracker.track(frame)
>         if 'polygon' in outputs:
>             polygon = np.array(outputs['polygon']).astype(np.int32)
>             cv2.polylines(frame, [polygon.reshape((-1, 1, 2))],
>                           True, (0, 255, 0), 3)
>             mask = ((outputs['mask'] > cfg.TRACK.MASK_THERSHOLD) * 255)
>             mask = mask.astype(np.uint8)
>             mask = np.stack([mask, mask*255, mask]).transpose(1, 2, 0)
>             frame = cv2.addWeighted(frame, 0.77, mask, 0.23, -1)
>         else:
>             bbox = list(map(int, outputs['bbox']))
>             cv2.rectangle(frame, (bbox[0], bbox[1]),
>                           (bbox[0]+bbox[2], bbox[1]+bbox[3]),
>                           (0, 255, 0), 3)
>         #cv2.imshow(video_name, frame)
>         #cv2.waitKey(40)
> 
> print(count / (time.time() - start_time))
> ```

Can you share the video? 
What is your CUDA, CuDNN and PyTorch version?

**Comment by ZhiyuanChen** [comment]
Since no futrher question is asked, I'm closing this issue for now. 
Please create a new issue should you have other questions.
